docs/jwt_profile.rst

```diff
@@ -11,26 +11,39 @@ All definitions are as per the RFCs unless otherwise stated.
 
 Note: All times will be UTC.
 
+GUID Definition
+===============
+All references to GUIDs refer to GUID (version 4); 128-bits in length as defined
+in RFC 4122 in their textual representation as defined in section 3
+"Namespace Registration Template" without the "urn:uuid:" prefix e.g.
+"f81d4fae-7dec-11d0-a765-00a0c91e6bf6".
+
+All GUIDs **MUST** be randomly generated such that there is negligible probability
+that the same value will be used twice, even across multiple servers. The same
+GUID value **MUST NOT** appear twice in the same JWT in any claim.
+
 JWS Protected Header
 ====================
-JWS Protected Header SHALL contain the following claims:
+JWS Protected Header **SHALL** contain the following claims:
 
 * 'typ' will be set to 'JWT'
 * 'alg' will be set to 'RS256'
 * 'kid' will be set to an appropriate value for the keys in use.
 
 JWT Payload
 ===========
-All JWT payloads SHALL include the follow claims:
+All JWT payloads **SHALL** include the follow claims:
+  'tx_id' - Set to a random GUID.
+    - Transaction ID used to trace a transaction through the whole system.
+    - **MUST NOT** be the same as the 'jti' value if present.
 
-    'tx_id'
-        Transaction ID used to trace a transaction through the whole system.
-        This will be a GUID (version 4) and 128-bits in length as defined in RFC 4122 in
-        its textual representation as defined in section 3 "Namespace Registration
-        Template" without the "urn:uuid:" prefix e.g.
-        "f81d4fae-7dec-11d0-a765-00a0c91e6bf6".
 
+All JWT payloads **MAY** contain the follow claims:
```

Comment:
Would the wording `All JWT payloads **SHOULD** contain the follow claims:` be better as going forward all tokens should contain the jti claim?